src/start/AuthScreen.js

```diff
@@ -251,16 +293,24 @@ class AuthScreen extends PureComponent<Props> {
           {activeAuthentications(
             serverSettings.authentication_methods,
             serverSettings.external_authentication_methods,
-          ).map(auth => (
-            <ZulipButton
-              key={auth.name}
-              style={styles.halfMarginTop}
-              secondary
-              text={`Log in with ${auth.displayName}`}
-              Icon={auth.Icon}
-              onPress={() => this.handleAuth(auth)}
-            />
-          ))}
+          ).map(auth =>
+            auth.name === 'apple' && Platform.OS === 'ios' ? (
+              <IosCompliantAppleAuthButton
+                key={auth.name}
+                style={styles.halfMarginTop}
+                onPress={() => this.handleAuth(auth)}
+              />
+            ) : (
+              <ZulipButton
+                key={auth.name}
+                style={styles.halfMarginTop}
+                secondary
+                text={`Sign in with ${auth.displayName}`}
```

Comment:
Small issue independent of this PR, but I notice it as I look at this code and particularly the set of translated strings at the same time:

This can't work when `auth.displayName` isn't one of the values we know about in advance, so in particular with an open-ended SSO like SAML. We should instead make this a fixed string with a placeholder, and pass the value of `auth.displayName` in through `values`.

Reply:
Ah, right—I should address this in my next revision, or as a separate PR (it didn't make it into the current revision).